### PR TITLE
chore: Fix linter findings for `revive:exported` in `plugins/inputs/[a-b]*`

### PR DESCRIPTION
**Pull request by zak-pawel**
## Summary

Address findings for [revive:exported ](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#exported ) in `plugins/inputs/[a-b]*`.

As part of this effort for files from `plugins/inputs/[a-b]*`, the following actions were taken:
- Type names (`const`, `var`, `struct`, `func`, etc) were changed to unexported, wherever they didn't need to be exported.
- All remaining exported types were given comments in the appropriate form – this does not apply to exported methods that implement "known" plugin interfaces (`Gather|Init|Start|Stop|SampleConfig|Parse|Add|Apply|Serialize|SerializeBatch|SetParser`).
- The order of methods was organized (exported methods first, then unexported, with `init` at the very end).

It is only part of the bigger job (for issue: https://github.com/influxdata/telegraf/issues/15813).
After all findings of this type in whole project are handled, we can enable `revive:exported ` rule in `golangci-lint`.

## Checklist

- [x] No AI generated code was used in this PR

### CODE REVIEW

File: plugins/inputs/activemq/activemq.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 //go:embed sample.conf
 var sampleConfig string
 
+// ActiveMQ Input Plugin
```
Comment (srebhan) [inline plugins/inputs/activemq/activemq.go:25]: Please don't add these nonsense comments!

File: plugins/inputs/aerospike/aerospike.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ import (
 //go:embed sample.conf
 var sampleConfig string
 
+// Aerospike Input Plugin
```
Comment (srebhan) [inline plugins/inputs/aerospike/aerospike.go:24]: Same here (and potentially everywhere else ;-)) What information does this comment add?

File: plugins/inputs/amqp_consumer/amqp_consumer.go [hunk 3]
```diff
@@ -62,11 +64,12 @@ type AMQPConsumer struct {
...
-
+// Mechanism returns "EXTERNAL"
 func (a *externalAuth) Mechanism() string {
 	return "EXTERNAL"
 }
+
+// Response returns "\000"
 func (a *externalAuth) Response() string {
 	return "\000"
 }
```
Comment (srebhan) [inline plugins/inputs/amqp_consumer/amqp_consumer.go:75]: Please revert this! Keep the struct close to the method definition of that struct and remove those comments telling you exactly the same as the code does!

File: plugins/inputs/aurora/aurora.go [hunk 4]
```diff
@@ -207,7 +208,7 @@ func (a *Aurora) gatherScheduler(
 		return fmt.Errorf("%v", resp.Status)
 	}
 
-	var vars Vars
+	var vars vars
```
Comment (srebhan) [inline plugins/inputs/aurora/aurora.go:211]: Ouch... This really hurts my eyes. :-)

**Review by srebhan — COMMENTED** [review]
Thanks for your effort @zak-pawel! I do have some comments with the most sever being to remove the useless comments! :-)

**Comment by zak-pawel** [comment]
> I do have some comments with the most sever being to remove the useless comments! :-)

@srebhan
The thing is, the `exported` rule is designed to detect uncommented "things" of the exported type. And while we can filter out methods with known names, it will be hard for us to do that for several hundred plugins and their public structs.

The options I see are:
1. Grit our teeth and put a comment like `// StructureName` over every public structure (representing a certain plugin) everywhere (we don't need anything more).
2. As above, but put `//nolint` xD
3. Did you know that these structs don't have to be exported? We can make them unexported... Only their fields need to be exported.
4. We can take the same steps I took in this PR for the entire codebase (except for the structs we're talking about), but not enable the `exported` rule.

Point 2 seems very poor. 
Similarly, point 4 - because I see value in this rule - it will make it easier for us to review new code, in which we often overlook that there are types that could be unexported.

**Comment by srebhan** [comment]
@zak-pawel how about adding a [config option to the linter](https://github.com/mgechev/revive/blob/798ce218499047f611481fe34c543acd21db2d75/rule/exported.go#L324)? :-)

**Comment by zak-pawel** [comment]
> @zak-pawel how about adding a [config option to the linter](https://github.com/mgechev/revive/blob/798ce218499047f611481fe34c543acd21db2d75/rule/exported.go#L324)? :-)

@srebhan To disable checking whether comments exist on the exported struct?

**Comment by srebhan** [comment]
Yes. We just need a PR on revive for that. :-P

**Comment by zak-pawel** [comment]
> Yes. We just need a PR on revive for that. :-P

@srebhan  Such PR already exists: https://github.com/mgechev/revive/pull/1047

File: plugins/inputs/activemq/activemq.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/inputs/activemq/activemq.go]: OK :(

File: plugins/inputs/aerospike/aerospike.go (hunk 2 above)
Comment (zak-pawel) [reply plugins/inputs/aerospike/aerospike.go]: Changed

File: plugins/inputs/amqp_consumer/amqp_consumer.go (hunk 3 above)
Comment (zak-pawel) [reply plugins/inputs/amqp_consumer/amqp_consumer.go]: Changed

File: plugins/inputs/aurora/aurora.go (hunk 4 above)
Comment (zak-pawel) [reply plugins/inputs/aurora/aurora.go]: I think it's beautiful ;)

File: plugins/inputs/bcache/bcache.go [hunk 5]
```diff
@@ -20,8 +20,44 @@ import (
 var sampleConfig string
 
 type Bcache struct {
-	BcachePath string
-	BcacheDevs []string
+	BcachePath string   `toml:"bcachePath"`
+	BcacheDevs []string `toml:"bcacheDevs"`
```
Comment (srebhan) [inline plugins/inputs/bcache/bcache.go:24]: I would prefer
```suggestion
	BcachePath string   `toml:"bcache_path"`
	BcacheDevs []string `toml:"bcache_devs"`
```

**Review by srebhan — COMMENTED** [review]
Nice, just one comment regarding the toml-option naming.

File: plugins/inputs/bcache/bcache.go (hunk 5 above)
Comment (zak-pawel) [reply plugins/inputs/bcache/bcache.go:24]: @srebhan Yes, me too. But I don't want to change names of current properties (look at `sample.conf` and `README`).

Comment (srebhan) [reply plugins/inputs/bcache/bcache.go:24]: I'm pretty sure it works both ways. Let's keep your version for now, but check if actually both works and change all the options instances to snake-case in a separate PR. Actually something we could use a linter for:

All exported fields of an exported struct _must_ have a `toml` annotation in snake_case. :-)

**Review by srebhan — APPROVED** [review]
Thanks @zak-pawel!

File: plugins/inputs/bcache/bcache.go (hunk 5 above)
Comment (zak-pawel) [reply plugins/inputs/bcache/bcache.go:24]: @srebhan 
After changing toml annotations to:
```
type Bcache struct {
	BcachePath string   `toml:"bcache_path"`
	BcacheDevs []string `toml:"bcache_devs"`
}
```

Telegraf fails to start:
```
2024-10-09T09:16:53Z I! Loading config: telegraf_bcache.conf
2024-10-09T09:16:53Z E! loading config file telegraf_bcache.conf failed: plugin inputs.bcache: line 157: configuration specified the fields ["bcachePath" "bcacheDevs"], but they were not used. This is either a typo or this config option does not exist in this version.
```

**Pull request merged**